### PR TITLE
Week-based UI, touch friendly lists, event recurrence updates

### PR DESCRIPTION
**Pull request by mgreen-simplethread**
This reshuffles the UI to look and work closer to the wireframe I've been working off of:

Screenshot 2020-03-19 14 17 34

I've also developed a touch-friendly list item component for use in the transaction details list. I'm still working on making it more accessible for people with motor and visual problems, because I anticipate that being a feedback item from CFPB eventually. Eventually I think I'll just make it slide open when it's tapped or long pressed. I'm open to suggestions.

The calendar view is now navigated in 1 week increments, and the details list shows all the transactions for that particular week.

When edited, recurring events now pop a dialog asking the user if they want to update the total of all future recurrences too, or just the event they're currently editing.

week-ui-demo

This is up on staging to demo interactively: https://cfgov-refresh-staging.herokuapp.com/mmt-my-money-calendar/calendar

### CODE REVIEW

File: cfgov/unprocessed/apps/mmt-my-money-calendar/css/components/calendar.less [hunk 1]
```diff
@@ -108,9 +146,76 @@
...
+  &__nav-button {
+    font-size: 24px;
+    background: none;
+    border: 0;
+
+    &:first-of-type {
+      margin-right: 10px;
+    }
+
+    &::last-of-type {
```
Comment (meissadia) [inline cfgov/unprocessed/apps/mmt-my-money-calendar/css/components/calendar.less]:
```suggestion
    &:last-of-type {
```

File: cfgov/unprocessed/apps/mmt-my-money-calendar/js/components/slide-list.js [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+}) {
+  const bem = useBEM('slide-list-item');
+  const rootClasses = clsx(bem(), className);
+  const background = useRef(null);
+  const foreground = useRef(null);
+  const isDragging = useRef(false);
+  const isOpen = useRef(false);
+  const slideWidth = useRef(0);
+  const [{ x }, set] = useSpring(() => ({ x: 0 }), { immediate: false });
+  //const transform = interpolate([x], (x) => `translateX(${x}px)`);
```
Comment (meissadia) [inline cfgov/unprocessed/apps/mmt-my-money-calendar/js/components/slide-list.js]:
```suggestion
```

File: cfgov/unprocessed/apps/mmt-my-money-calendar/css/components/calendar.less (hunk 1 above)
Comment (mgreen-simplethread) [reply cfgov/unprocessed/apps/mmt-my-money-calendar/css/components/calendar.less]: This is a pseudoclass, not a pseudoelement. I think 1 colon is right in this case.

Comment (mgreen-simplethread) [reply cfgov/unprocessed/apps/mmt-my-money-calendar/css/components/calendar.less]: Nevermind, I read the diff wrong. This is indeed a typo

File: cfgov/unprocessed/apps/mmt-my-money-calendar/js/components/slide-list.js (hunk 2 above)
Comment (mgreen-simplethread) [reply cfgov/unprocessed/apps/mmt-my-money-calendar/js/components/slide-list.js]: Fixed!

File: cfgov/unprocessed/apps/mmt-my-money-calendar/css/components/calendar.less (hunk 1 above)
Comment (mgreen-simplethread) [reply cfgov/unprocessed/apps/mmt-my-money-calendar/css/components/calendar.less]: Fixed

**Comment by meissadia** [comment]
Small bug when the start of a week falls in the previous month.  The calendar view does not update.  Happens again when going from August -> September.

May-June

**Comment by meissadia** [comment]
Selecting 'Recurring' throws off page alignment.  This does not happen on mobile.
recurring-out-of-view

**Comment by meissadia** [comment]
`NaN`s when there's no data to calculate from.

Screen Shot 2020-03-19 at 1 50 21 PM

**Comment by mgreen-simplethread** [comment]
@meissadia Thanks for catching these issues! I've fixed everything except for the desktop shift bug. I'm going to do a desktop pass once I've got a completely finalized UI design for it and the rest of mobile, and I'll address it then.

**Pull request merged**